a11ym

```diff
@@ -250,4 +263,43 @@ crawler.on('fetchcomplete', function(queueItem) {
     urls.push({url: url});
 });
 
-crawler.start();
+var computeURLToCrawler = function (crawler) {
+    var first = true;
+
+    return function (value) {
+        var url = crawler.processURL(value);
+        console.log(url);
+
+        if (true === first) {
+            first                   = false;
+            crawler.host            = url.host;
+            crawler.initialProtocol = url.protocol;
+            crawler.initialPort     = url.port;
+            crawler.initialPath     = url.path;
+        } else {
+            ++maximumUrls;
+        }
+
+        crawler.queue.add(
+            url.protocol,
+            url.host,
+            url.port,
+            url.path
+        );
+    };
+};
+
+if (1 === program.args.length && '-' === program.args[0]) {
+    crawler.maxDepth = 1;
```

Comment:
Is this really what we want ?

It could make sense to first crawl the given URLs and then start crawling the ones found in them. Not sure though, maybe wait and see ?